unix/man/man1/blaze.1
.\" manual page for blaze, a command wrapper for BlazeBlogger
.\" Copyright (C) 2010-2011 Jaromir Hradilek
.\"
.\" Permission is granted to copy, distribute and/or modify this document
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation;  with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.
.\"
.\" A copy  of the license is included  as a file called FDL  in the main
.\" directory of the BlazeBlogger source package.
.TH BLAZE 1 "2011-02-18" "Version 1.2.0" "BlazeBlogger Documentation"
.SH NAME
blaze \- allows you to run BlazeBlogger utilities with a single command
.SH SYNOPSIS
.B  blaze
.I  command
.RI [ option ...]
.PP
.B  blaze
.BR help | man
.RI [ command ]
.PP
.B blaze
.BR \-h | \-v
.SH DESCRIPTION
.B blaze
provides a common way to run BlazeBlogger utilities with a single command.
.SH OPTIONS
.SS Command Line Options
.TP
.BR \-h ", " \-\-help
Displays usage information and exits.
.TP
.BR \-v ", " \-\-version
Displays version information and exits.
.SS Available Commands
.TP
.BR init ", " in
Runs
.BR blaze-init ,
a utility that either creates a BlazeBlogger repository, or recovers a
previously corrupted one.
.TP
.BR config ", " cf ", " cfg
Runs
.BR blaze-config ,
a utility that either displays, or sets BlazeBlogger configuration options.
.TP
.B  add
Runs
.BR blaze-add ,
a utility that adds a blog post or a page to the BlazeBlogger repository.
.TP
.BR edit ", " ed
Runs
.BR blaze-edit ,
a utility that edits a blog post or a page in the BlazeBlogger repository.
.TP
.BR remove ", " rm ", " del
Runs
.BR blaze-remove ,
a utility that removes a blog post or a page from the BlazeBlogger
repository.
.TP
.BR list ", " ls
Runs
.BR blaze-list ,
a utility that lists blog posts or pages in the BlazeBlogger repository.
.TP
.BR make ", " mk
Runs
.BR blaze-make ,
a utility that generates a blog from the BlazeBlogger repository.
.TP
.B  log
Runs
.BR blaze-log ,
a utility that displays the BlazeBlogger repository log.
.TP
.B  help
Displays usage information, either for the utility itself, or for the
selected
.IR command .
.TP
.B  man
Displays a manual page, either for the utility itself, or for the selected
.IR command .
.TP
.BR version ", " vs ", " ver
Displays version information.
.SH ENVIRONMENT
.TP
.B  EDITOR
Unless the
.B  core.editor
option is set,
.B  blaze
tries to use system-wide settings to decide which editor to use.
.SH SEE ALSO
.BR blaze-init (1),
.BR blaze-config (1),
.BR blaze-add (1),
.BR blaze-edit (1),
.BR blaze-remove (1),
.BR blaze-list (1),
.BR blaze-make (1),
.BR blaze-log (1)
.SH BUGS
To report a bug or to send a patch, please, add a new issue to the bug
tracker at <http://code.google.com/p/blazeblogger/issues/>, or visit the
discussion group at <http://groups.google.com/group/blazeblogger/>.
.SH COPYRIGHT
Copyright (C) 2009-2011 Jaromir Hradilek
.PP
This program is free software; see the source for copying conditions. It is
distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.


unix/blaze.sh
#!/bin/sh

# blaze, a command wrapper for BlazeBlogger
# Copyright (C) 2009-2011 Jaromir Hradilek

# This program is  free software:  you can redistribute it and/or modify it
# under  the terms  of the  GNU General Public License  as published by the
# Free Software Foundation, version 3 of the License.
#
# This program  is  distributed  in the hope  that it will  be useful,  but
# WITHOUT  ANY WARRANTY;  without  even the implied  warranty of MERCHANTA-
# BILITY  or  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
# License for more details.
#
# You should have received a copy of the  GNU General Public License  along
# with this program. If not, see <http://www.gnu.org/licenses/>.

# General script information:
NAME=${0##*/}
VERSION='1.2.0'

# Get the command if any, and shift command line options:
COMMAND=$1
shift

# Substitute aliases:
case "$COMMAND" in
  "ed")         COMMAND="edit";;
  "in")         COMMAND="init";;
  "ls")         COMMAND="list";;
  "mk")         COMMAND="make";;
  "rm" | "del") COMMAND="remove";;
  "cf" | "cfg") COMMAND="config";;
  "vs" | "ver") COMMAND="version";;
esac

# Parse the command and perform an appropriate action:
case "$COMMAND" in
  "add" | "log" | "edit" | "init" | "list" | "make" | "config" | "remove")
    # Run the selected utility:
    exec blaze-$COMMAND "$@"
    ;;
  "-h" | "--help" | "help")
    # Get the command, if any:
    COMMAND=$1

    # Parse the command, and display its usage:
    case "$COMMAND" in
      "add" | "log" | "edit" | "init" | "list" | "make" | "config" | "remove")
        # Display the utility usage information:
        exec blaze-$COMMAND --help
        ;;
      *)
        # Display the list of available commands:
        echo "Usage: $NAME COMMAND [OPTION...]"
        echo
        echo "Basic commands:"
        echo "  init     create or recover a BlazeBlogger repository"
        echo "  config   display or set BlazeBlogger configuration options"
        echo "  add      add a blog post or page to a BlazeBlogger repository"
        echo "  edit     edit a blog post or page in a BlazeBlogger repository"
        echo "  remove   remove a blog post or page from a BlazeBlogger repository"
        echo "  list     list blog posts or pages in a BlazeBlogger repository"
        echo "  make     generate a blog from a BlazeBlogger repository"
        echo "  log      display a BlazeBlogger repository log"
        echo
        echo "Additional commands:"
        echo "  help [COMMAND]  display usage information on the selected command"
        echo "  man [COMMAND]   display a manual page for the selected command"
        echo "  version         display version information"
        echo
        echo "Command aliases:"
        echo "  in       init"
        echo "  ed       edit"
        echo "  ls       list"
        echo "  mk       make"
        echo "  cf, cfg  config"
        echo "  rm, del  remove"
        echo "  vs, ver  version"

        # Return success:
        exit 0
        ;;
    esac
    ;;
  "man")
    # Get the command, if any:
    COMMAND=$1

    # Parse the command, and display its manual page:
    case "$COMMAND" in
      "add" | "log" | "edit" | "init" | "list" | "make" | "config" | "remove")
        # Display the utility usage information:
        exec man blaze-$COMMAND
        ;;
      *)
        # Display a general manual page:
        exec man blaze
        ;;
    esac
    ;;
  "-v" | "--version" | "version")
    # Display version information:
    echo "BlazeBlogger $VERSION"
    echo
    echo "Copyright (C) 2008-2011 Jaromir Hradilek"
    echo "This program is free software; see the source for copying conditions. It is"
    echo "distributed in the hope  that it will be useful,  but WITHOUT ANY WARRANTY;"
    echo "without even the implied warranty of  MERCHANTABILITY or FITNESS FOR A PAR-"
    echo "TICULAR PURPOSE."

    # Return success:
    exit 0
    ;;
  *)
    # Respond to a wrong or missing command:
    echo "Usage: $NAME COMMAND [OPTION...]" >&2
    echo "Try \`$NAME help' for more information." >&2

    # Return failure:
    exit 22
    ;;
esac
